api/apps/api/src/modules/specification/determine-features/application/determine-features-for-scenario.handler.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  CommandHandler,
+  EventPublisher,
+  IInferredCommandHandler,
+} from '@nestjs/cqrs';
+import { Logger } from '@nestjs/common';
+
+import { SpecificationRepository } from './specification.repository';
+import { DetermineFeaturesForSpecification } from '../../commands/determine-features-for-scenario-specification.command';
+
+@CommandHandler(DetermineFeaturesForSpecification)
+export class DetermineFeaturesForScenarioHandler
+  implements IInferredCommandHandler<DetermineFeaturesForSpecification> {
+  private readonly logger: Logger = new Logger(
+    DetermineFeaturesForSpecification.name,
+  );
+
+  constructor(
+    private readonly eventPublisher: EventPublisher,
+    private readonly specificationRepository: SpecificationRepository,
+  ) {}
+
+  async execute({
+    featuresConfig,
+    specificationId,
+  }: DetermineFeaturesForSpecification): Promise<void> {
+    let specification = await this.specificationRepository.getById(
+      specificationId,
+    );
+    if (!specification) {
+      this.logger.warn(`Couldn't find specification: ${specificationId}`);
+      return;
+    }
+    specification = this.eventPublisher.mergeObjectContext(specification);
+
+    specification.determineFeatures([featuresConfig]);
+    await this.specificationRepository.save(specification);
```

Comment:
shouldn't the specification be found by features ids? Also, should we use a transaction/pesimistic lock here on specifications?

Reply:
1 I decided to separate the `find by feature ids` to standalone application as it will be reused by both `determineFeatures` and `calculateFeatures`
2) we could, do you think there is a high chance it will clash?

Reply:
1. what standalone application do you mean? The repo Is a driven port here, so the repository implementation is not a part of application but it's the adapter
2. when the queue will calculate features one by one then yes, it may cause lost updates

Reply:
1. Sorry, I meant standalone handler using different (repo)port which will dispatch commands for each found scenario (spec)